joseTOV/metamodel_eos.py

```diff
@@ -0,0 +1,521 @@
+        v_sym2_1 = self.L_sym -     self.t_sat * ( 2**(2/3)*( 2+5*self.kappa_NM+8* self.kappa_NM2+ 11*self.kappa_NM3) - ( 2+5*self.kappa_sat+8* self.kappa_sat2+11*self.kappa_sat3)  ) - v_nq[1]
+        v_sym2_2 = self.K_sym - 2 * self.t_sat * ( 2**(2/3)*(-1+5*self.kappa_NM+20*self.kappa_NM2+ 44*self.kappa_NM3) - (-1+5*self.kappa_sat+20*self.kappa_sat2+44*self.kappa_sat3)  ) - v_nq[2]
+        v_sym2_3 = self.Q_sym - 2 * self.t_sat * ( 2**(2/3)*( 4-5*self.kappa_NM+40*self.kappa_NM2+ 220*self.kappa_NM3) - ( 4-5*self.kappa_sat+40*self.kappa_sat2+220*self.kappa_sat3) ) - v_nq[3]
+        v_sym2_4 = self.Z_sym - 8 * self.t_sat * ( 2**(2/3)*(-7+5*self.kappa_NM-10*self.kappa_NM2+ 110*self.kappa_NM3) - (-7+5*self.kappa_sat-10*self.kappa_sat2+110*self.kappa_sat3) ) - v_nq[4]
+        
+        self.v_sym2 = jnp.array([v_sym2_0, v_sym2_1, v_sym2_2, v_sym2_3, v_sym2_4])
+        
+                
+        ### TODO: empty crust not used for now
+        # if use_empty_crust:
+        #     ns_crust, ps_crust, es_crust = jnp.array([]), jnp.array([]), jnp.array([])
+        # else:
+        
+        # Load and preprocess the crust
+        ns_crust, ps_crust, es_crust = load_crust(crust)
+        mask = ns_crust <= max_n_crust
+        ns_crust, ps_crust, es_crust = ns_crust[mask], ps_crust[mask], es_crust[mask]
+        mu_lowest = (es_crust[0] + ps_crust[0]) / ns_crust[0]
+        
+        print("WARNING OVERRIDING FOR DEBUGGING")
+        mu_lowest = 930.1193490245807
+        print("Mu lowest jose:", mu_lowest)
+        
+        cs2_crust = jnp.gradient(ps_crust, es_crust)
+
+        # Make sure the metamodel starts above the crust
+        max_n_crust = ns_crust[-1]
+        nmin = max(nmin, max_n_crust + 1e-3)
+        self.nmin = nmin
+        
+        # Create the density array
+        self.nmax = nmax_nsat * self.nsat
+        self.ndat = ndat
+        n_metamodel = jnp.linspace(nmin, self.nmax, ndat)
+        
+        # Get cs2 for the metamodel
+        cs2_metamodel = self.cs2_tot(n_metamodel)
+        
+        # Spline for speed of sound for the connection region
+        ns_spline = jnp.append(ns_crust, n_metamodel)
+        cs2_spline = jnp.append(cs2_crust, cs2_metamodel)
+        
+        n_connection = jnp.linspace(max_n_crust, self.nmin, ndat_spline)
+        cs2_connection = utils.cubic_spline(n_connection, ns_spline, cs2_spline)
+        cs2_connection = jnp.clip(cs2_connection, 1e-5, 1.0)
+        
+        # Concatenate the arrays
+        n = jnp.concatenate([ns_crust, n_connection, n_metamodel])
+        cs2 = jnp.concatenate([cs2_crust, cs2_connection, cs2_metamodel])
+        
+        # Compute pressure and energy and initialize the parent class with it
+        # TODO: why here only mu lowest?
+        mu = jnp.ones(len(n)) * mu_lowest
```

Comment:
Here, the `mu_lowest` returns and `mu` is a constant array equal to its value. Therefore, this has a large impact on the calculation and final values of the pressure and energy. This seems a bit weird to me. Also, can't we compute `mu` instead of fixing it to this specific value?